src/experimental/null_allocator.zig

```diff
@@ -0,0 +1,65 @@
+// The NullAllocator is meant to be used as a fallback
+// for allocator composition.
+
+// For example, a StackAllocator, if it runs out of memory,
+// can dispatch to another allocator to fulfill the request.
+// If we want that to signal an error, we can give it the
+// NullAllocator as a fallback and that will signal an
+// "OutOfMemory" error, enforcing that we don't ask for
+// more memory than is on the stack.
```

Comment:
This is actually pretty genius!

Reply:
It's also possible to use this for something like an allocator that can be "turned off" by mutating itself and replacing its composed allocator with this allocator.

Reply:
Huh, now that's an interesting idea. Since we can access the member variables directly in Zig, I think this is something that we can already express?